app/controllers/sport_sessions_controller.rb

```diff
@@ -4,18 +4,25 @@ class SportSessionsController < ApplicationController
 
   def index
     @sport_sessions = policy_scope(SportSession)
-    @markers = @sport_sessions.geocoded.map do |sport_session|
+    @venues = @sport_sessions.map(&:venue)
+
+    @markers = @venues.map do |venue|
       {
-        lat: sport_session.latitude,
-        lng: sport_session.longitude
+        lat: venue.latitude,
+        lng: venue.longitude
       }
     end
   end
 
   def show
     @full = @sport_session.max_attendees == @sport_session.attendees.count
     authorize @sport_session
-    @marker = { lat: @sport_session.latitude, lng: @sport_session.longitude }
+    @markers = [
```

Comment:
Change here was to markets, while show is only 1 item, the maps view to keep simplified works on the basis that the data is an array. By making it an array here we simplify our code.